test/milton/MiltonSoap.t.sol

```diff
@@ -0,0 +1,1903 @@
+            address(_daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(mockCase0StanleyDai)
+        );
+        MockCase0JosephDai mockCase0JosephDai = getMockCase0JosephDai(
+            address(_daiMockedToken),
+            address(_ipTokenDai),
+            address(mockCase0MiltonDai),
+            address(miltonStorageDai),
+            address(mockCase0StanleyDai)
+        );
+        int256 expectedSoap75Days = TestConstants.ZERO_INT;
+        prepareApproveForUsersDai(_users, _daiMockedToken, address(mockCase0JosephDai), address(mockCase0MiltonDai));
+        prepareMilton(mockCase0MiltonDai, address(mockCase0JosephDai), address(mockCase0StanleyDai));
+        prepareJoseph(mockCase0JosephDai);
+        prepareIpToken(_ipTokenDai, address(mockCase0JosephDai));
+        _miltonSpreadModel.setCalculateQuoteReceiveFixed(TestConstants.PERCENTAGE_2_18DEC);
+        vm.prank(_liquidityProvider);
+        mockCase0JosephDai.itfProvideLiquidity(2 * TestConstants.USD_28_000_18DEC, block.timestamp);
+        // when
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(_daiMockedToken), TestConstants.PERCENTAGE_6_18DEC, block.timestamp);
+        openSwapReceiveFixed(
+            _userTwo,
+            block.timestamp,
+            1000348983489384893923,
+            TestConstants.PERCENTAGE_1_18DEC,
+            TestConstants.LEVERAGE_1000_18DEC,
+            mockCase0MiltonDai
+        );
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(
+            address(_daiMockedToken),
+            TestConstants.PERCENTAGE_3_18DEC,
+            block.timestamp + TestConstants.PERIOD_25_DAYS_IN_SECONDS
+        );
+        openSwapReceiveFixed(
+            _userTwo,
+            block.timestamp + TestConstants.PERIOD_25_DAYS_IN_SECONDS,
+            1492747383748202058744,
+            TestConstants.PERCENTAGE_1_18DEC,
+            TestConstants.LEVERAGE_1000_18DEC,
+            mockCase0MiltonDai
+        );
+        vm.startPrank(_userOne);
+        iporOracle.itfUpdateIndex(
+            address(_daiMockedToken),
+            TestConstants.PERCENTAGE_3_18DEC,
+            block.timestamp + TestConstants.PERIOD_50_DAYS_IN_SECONDS
+        );
+        mockCase0MiltonDai.itfCloseSwapReceiveFixed(1, TestConstants.PERIOD_75_DAYS_IN_SECONDS);
+        mockCase0MiltonDai.itfCloseSwapReceiveFixed(2, TestConstants.PERIOD_75_DAYS_IN_SECONDS);
+        vm.stopPrank();
+        // then
+        (,, int256 soap75Days) =
+            calculateSoap(_userTwo, block.timestamp + TestConstants.PERIOD_75_DAYS_IN_SECONDS, mockCase0MiltonDai);
+        assertEq(soap75Days, expectedSoap75Days);
+    }
+
+    function testShouldCalculateSoapEqualZeroWhenDAIReceiveFixedAtIPOR6PercentAndIPOR6PercentAfter25DaysAndDAIReceiveFixedAndIPOR3PercentAfter50DaysAndCloseAllSwapsAfter75Days(
```

Comment:
This test is not exists in old tests and this test has wrong description.
Part "AndIPOR6PercentAfter25Days" - in line 1663 you update index using 3% not 6%

I suggest to remove this test.